docs/QuickStart.md

```diff
@@ -0,0 +1,204 @@
+auto sort(spider::Context &context, int x, int y) -> std::tuple<int, int> {
+    if (x >= y) {
+        return { x, y };
+    }
+    return { y, x };
+}
+```
+
+## Run a Task
+Spider enables user to run a task on the cluster. First register the functions statically so it is known by Spider. Simply call `Driver::run` and provide the arguments of the task. `Driver::run` returns a `spider::Future` object, which represents the result that will be available in the future. You can call `Future::ready` to check if the value in future is available yet. You can use `Future::get` to block and get the value once it is available. 
+```c++
+spider::register_task(sum);
+spider::register_task(sort);
+
+auto main(int argc, char **argv) -> int {
+    // driver initialization skipped
+    spider::Future<int> sum_future = driver.run(sum, 2);
+    assert(4 == sum_future.get());
+
+    spider::Future<std::tuple<int, int>> sort_future = driver.run(4, 3);
+    assert(std::tuple{3, 4} == sort_future.get());
+}
+```
+
+## Group Tasks Together
+In real world, running a single task is too simple to be useful. Spider lets you bind outputs of tasks as inputs of another task, similar to `std::bind`. Binding the  tasks together forms a dependencies among tasks, which is represented by `spider::TaskGraph`. `TaskGraph` can be further bound into more complicated `TaskGraph` by serving as inputs for another task. You can run the task using `Driver::run` in the same way as running a single task.
+```c++
+auto square(spider::Context& context, int x) -> int {
+    return x * x;
+}
+
+auto square_root(spider::Context& context, int x) -> int {
+    return sqrt(x);
+}
+// task registration skipped
+auto main(int argc, char **argv) -> auto {
+    // driver initialization skipped
+    spider::TaskGraph<int(int, int)> sum_of_square = spider::bind(sum, square, square);
+    spider::TaskGraph<int(int, int)> rss = spider::bind(square_root, sum_of_square);
+    spider::Future<int> future = driver::run(rss, 3, 4);
+    assert(5 == future.get());
+}
+```
+
+## Run Task inside Task
+Static task graph is enough to solve a lot of real work problems, but dynamically add tasks on-the-fly could become handy. As mentioned before, spider allows you to add another task as child of the running task by calling `Context::add_child`.
+
+```c++
+auto gcd(spider::Conect& context, int x, int y) -> std::tuple<int, int> {
+    if (x == y) {
+        std::cout << "gdc is: " << x << std::endl;
+        return { x, y };
+    }
+    if (x > y) {
+        context.add_child(gcd);
+        return { x % y, y };
+    }
+    context.add_child(gcd);
+    return { x, y % x };
+}
+```
+
+However, it is impossible to get the return value of the task graph from a client. We have a solution by sharing data using key-value store, which will be discussed later. Another solution is to run task or task graph inside a task and wait for its value, just like a client. This solution is closer to the conventional function call semantic.
```

Comment:
> However, it is impossible to get the return value of the task graph from a client.

This sentence doesn't make sense since the previous section has an example where the client is retrieving the result of a task graph.

Reading the later parts of this guide, I guess you mean that you can't return the value of a dynamically created task to the client (which is intuitive since the interface doesn't provide a mechanism to do so).